docker/android/android/make.py

```diff
@@ -0,0 +1,513 @@
+'''
```

Comment:
Because of how the Android Makefiles are designated, we don't need a complete or correct parser in earlier versions to separate the tests from the actual non-test targets. This is because all tests are designated by something like:

```Makefile
##################################
# test executable
```

Or:

```Makefile
# Unit tests.
# -------------------------------------
```

The valid section headers are `#########` (repeated more), `# -----------`, or `# =========`, and they can sandwich the header, be on top, or on bottom. The header title can also be multiple lines. We handle all these cases, including empty headers like:

```Makefile
#######################
#


```